venues/ICML.cc/2018/RML/webfield/programchairWebfield.js

```diff
@@ -0,0 +1,706 @@
+      }
+      numOfReviewers++;
+    }
+
+    var averageRating = 'N/A';
+    var minRating = 'N/A';
+    var maxRating = 'N/A';
+    if (ratings.length) {
+      averageRating = _.round(_.sum(ratings) / ratings.length, 2);
+      minRating = _.min(ratings);
+      maxRating = _.max(ratings);
+    }
+
+    if (ratings.length == numOfReviewers) {
+      numCompletedReviews++;
+    }
+
+
+    return {
+      note: paper.note,
+      averageRating: averageRating,
+      maxRating: maxRating,
+      minRating: minRating,
+      numOfReviews: ratings.length,
+      numOfReviewers: numOfReviewers,
+   }
+  });
+
+  var reviewProgressData = {
+    numCompletedReviews: numCompletedReviews,
+    numPapers: papers.length,
+    papers: _.sortBy(paperProgressData, [function(p) { return p.note.number; }])
+  }
+
+  return {
+    summary: summary,
+    reviewProgressData: reviewProgressData
+  }
+
+};
+
+
+var buildPCTableRow = function(index, reviewer, papers) {
+
+  var summary = {
+    id: reviewer.id,
+    name: reviewer.name,
+    email: reviewer.email
+  }
+
+  var reviewProgressData = {
+    numCompletedReviews: _.size(_.filter(papers, function(p) { return p.review; })),
+    numPapers: papers.length,
+    papers: _.sortBy(papers, [function(p) { return p.note.number; }])
+  }
+
+  var numCompletedReviews = 0;
+  var paperProgressData = _.map(papers, function(paper) {
+    var ratings = [];
+    var numOfReviewers = 0;
+
+    for (var reviewerNum in paper.reviewers) {
+      if (reviewerNum in paper.reviews) {
+        ratings.push(paper.reviews[reviewerNum].rating);
+      }
+      numOfReviewers++;
+    }
+
+    var averageRating = 'N/A';
+    var minRating = 'N/A';
+    var maxRating = 'N/A';
+    if (ratings.length) {
+      averageRating = _.round(_.sum(ratings) / ratings.length, 2);
+      minRating = _.min(ratings);
+      maxRating = _.max(ratings);
+    }
+
+    if (ratings.length == numOfReviewers) {
+      numCompletedReviews++;
+    }
+
+    return {
+      note: paper.note,
+      averageRating: averageRating,
+      maxRating: maxRating,
+      minRating: minRating,
+      numOfReviews: ratings.length,
+      numOfReviewers: numOfReviewers,
+    }
+  });
+
+  var reviewStatusData = {
+    numCompletedReviews: numCompletedReviews,
+    numPapers: papers.length,
+    papers: _.sortBy(paperProgressData, [function(p) { return p.note.number; }])
+  }
+
+  return {
+    summary: summary,
+    reviewProgressData: reviewProgressData,
+    reviewStatusData: reviewStatusData
+  }
+};
+
+var buildNoteMap = function(noteNumbers) {
+  var noteMap = Object.create(null);
+  for (var i = 0; i < noteNumbers.length; i++) {
+    noteMap[noteNumbers[i]] = Object.create(null);
+  }
+  return noteMap;
+};
+
+
+// Kick the whole thing off
+var headerLoaded = $.Deferred();
+displayHeader(headerLoaded);
+
+$.ajaxSetup({
+  contentType: 'application/json; charset=utf-8'
+});
+
+controller.addHandler('areachairs', {
```

Comment:
remove this code?

Reply:
I think this is a token used by the controller, we could rename it to 'programChairsConsole'?